src/expr.rs

```diff
@@ -53,11 +55,37 @@ impl Rewrite for ast::Expr {
             ast::Expr_::ExprTup(ref items) => {
                 rewrite_tuple_lit(context, items, self.span, width, offset)
             }
+            ast::Expr_::ExprLoop(ref block, _) => {
+                // FIXME: this drops any comment between "loop" and the block.
+                block.rewrite(context, width, offset).map(|result| {
+                    format!("loop {}", result)
+                })
+            }
             _ => context.codemap.span_to_snippet(self.span).ok()
         }
     }
 }
 
+impl Rewrite for ast::Block {
+    fn rewrite(&self, context: &RewriteContext, _: usize, _: usize) -> Option<String> {
+        let mut visitor = FmtVisitor::from_codemap(context.codemap, context.config);
+        visitor.last_pos = self.span.lo;
+        visitor.block_indent = context.block_indent;
+
+        visitor.visit_block(self);
+
+        // Push text between last block item and end of block
+        let snippet = visitor.snippet(mk_sp(visitor.last_pos, self.span.hi));
+        visitor.changes.push_str_span(self.span, &snippet);
+
+        // Stringify visitor
+        let file_name = context.codemap.span_to_filename(self.span);
+        let string_buffer = visitor.changes.get(&file_name);
```

Comment:
I'm not saying you need to make the changes as part of this PR, but just going to note some odd-ness about the FmtVisitor API here:

* We've just filled the visitor with local changes, and yet now we need a filename to get them out. Furthermore, we have to reach inside the visitor to its changes field to get them. I'd prefer some API to just dump everything collected so far into a String (or StringBuffer).

* With the calls to add any missed text, this seems like it should be an internal operation on the visitor - there should be a function which takes a position (self.span.hi, in this case), and does the rest.